src/desktop/dialogs/settingsdialog/contentfilter.cpp

```diff
@@ -0,0 +1,195 @@
+	stack->setContentsMargins(0, 0, 0, 0);
+	layout->addWidget(stack);
+
+	auto *formPage = new QWidget;
+	auto *form = new utils::SanerFormLayout(formPage);
+	form->setContentsMargins(0, 0, 0, 0);
+	form->setStretchLabel(false);
+	stack->addWidget(formPage);
+
+	auto *advisory = new QCheckBox(tr("Filter sessions with a content advisory"));
+	settings.bindContentFilterAdvisoryTag(advisory);
+	form->addRow("Advisory filtering:", advisory);
+	form->addRow(nullptr, utils::note(tr("Advisories are added by session owners "
+		"when they believe a session may not be safe for everyone."), QSizePolicy::CheckBox));
+
+	form->addSpacer();
+
+	auto *autoTag = new QCheckBox(tr("Filter sessions with titles containing:"));
+	form->addRow(tr("Keyword filtering:"), autoTag);
+	settings.bindContentFilterAutoTag(autoTag);
+	auto *tagWords = new QLineEdit(this);
+	tagWords->setPlaceholderText(contentfilter::defaultWordList());
+	utils::setSpacingControlType(tagWords, QSizePolicy::CheckBox);
+	form->addRow(nullptr, utils::indent(tagWords));
+	settings.bindContentFilterTags(tagWords);
+	settings.bindContentFilterAutoTag(tagWords, &QLineEdit::setEnabled);
+
+	form->addSpacer();
+
+	auto *level = form->addRadioGroup(tr("Filter mode:"), false, {
+		{ tr("Unrestricted"), int(contentfilter::Level::Unrestricted) },
+		{ tr("Remove sessions from listings"), int(contentfilter::Level::NoList) },
+		{ tr("Prevent joining filtered sessions"), int(contentfilter::Level::NoJoin) },
+		{ tr("Disconnect when sessions become not safe for me"), int(contentfilter::Level::Restricted) }
+	});
+	level->button(1)->setToolTip(tr("Sessions will be hidden from server "
+		"listings, but can still be joined by following a direct link."));
+	level->button(2)->setToolTip(tr("Sessions will be hidden from server "
+		"listings and cannot be joined."));
+	level->button(3)->setToolTip(tr("Sessions will be hidden from server "
+		"listings and cannot be joined. Connected sessions that change "
+		"their title or advisory flags will automatically disconnect."));
+	settings.bindContentFilterLevel(level);
+
+	auto *forceCensor = new QCheckBox(tr("Block layer uncensoring"));
+	settings.bindContentFilterForceCensor(forceCensor);
+	form->addRow(nullptr, forceCensor);
+
+	layout->addStretch();
+
+	auto *lockLayout = new QHBoxLayout;
+	lockLayout->setContentsMargins(0, 0, 0, 0);
+	layout->addLayout(lockLayout);
+
+	auto *lock = new QPushButton(tr("Lock"));
+	lock->setAutoDefault(false);
+	lockLayout->addWidget(lock);
+	connect(lock, &QPushButton::clicked, this, &ContentFilter::toggleLock);
+
+	auto *hide = new QCheckBox(tr("Hide settings when locked"));
+	lockLayout->addWidget(hide);
+	settings.bindContentFilterHideLocked(hide);
+
+	lockLayout->addStretch();
+
+	auto *hiddenPage = new QWidget;
+	auto *hiddenLayout = new QVBoxLayout(hiddenPage);
+	hiddenLayout->setContentsMargins(0, 0, 0, 0);
+	hiddenLayout->addWidget(new QLabel(tr("Content filtering settings are currently locked.")), 0, Qt::AlignCenter);
```

Comment:
"Content filtering" makes it sound like we filter the content of sessions, which we sure don't. Parental controls seems like a fine and common name for them. Dito for all uses of this term.